lib/developer_portal/app/controllers/developer_portal/admin/account/payment_details_base_controller.rb

```diff
@@ -28,7 +28,7 @@ def edit
   def update
     current_account.updating_payment_detail = true
     if current_account.update account_params
-      redirect_to payment_details_path, notice: 'Your billing address was successfully stored'
+      update_billing_address_on_stripe(account_params)
```

Comment:
Also, this is a bit hard to follow - I'd avoid having a redirect, and a duplicated error-rendering functionality in a method.
Instead, I would do something like this:

```
if update_address_on_gateway && current_account.update account_params
  redirect_to payment_details_path, notice: 'Your billing address was successfully stored'
else
  flash[:notice] ...
end
```

And then (in pseudocode):
```
def update_address_on_gateway
  return true if <payment gateway is not Stripe>

  <the code updating the address in Stripe, returning true or false>
end
```

But probably still better to move this code to the Stripe controller.